### DISCUSSION TITLE
Strategy for sharing devcontainers with coder containers

### DISCUSSION THREAD

**Discussion by geiseri** [coder/coder#5190] [General] [answered]
I am using the docker-image-builds template for using my devcontainer with code-server.  Right now, I need to manually sync the container from the project to the template.  This causes my DRY sensibilities to scream at me ;)

I tried to use the docker in docker approach, but it would complain about the bind mount for the project directory. 

Anyone else have some ideas here?

**Comment by jsjoeio** [comment]
That does not sound ideal! @bpmct any ideas?

**Comment by ghuntley** [comment]
> sharing devcontainers with coder containers
> docker-image-builds template

Hey @geiseri can you please gist up a rough mockup that replicates what your configuration looks like?

**Reply by geiseri** [reply]
@ghuntley 

After messing around with it, I figured out the bare minimum needed.  It looks like I need to be able to check out the source repository and then access the docker file in `.devcontainer/Dockerfile`.  Something like this:
```
resource "docker_image" "coder_image" {
  name = "coder-base-${data.coder_workspace.me.owner}-${lower(data.coder_workspace.me.name)}"
  build {
    context    = "./devcontainer"
    dockerfile = "./devcontainer/Dockerfile"
    tag        = ["coder-${data.coder_parameter.docker_image.value}:v0.1"]
  }
  # Keep alive for other workspaces to use upon deletion
  keep_locally = true
}
```

Instead of `./devcontainer` coming from the template repository, it would come from the source checkout.

My current workflow is to add the [docker-image-builds/main.tf](https://github.com/coder/coder/blob/main/examples/templates/docker-image-builds/main.tf) file to the source repository I would like to develop with and then push that as a template.  It's a bit hamfisted, but it's not horrible.

So I guess the "simplest thing that could possibly work" would be to get the docker build information from the source repository. 

Does that make more sense?

**Reply by bpmct** [reply]
We're working on native devcontainer support now, the UX should improve significantly within the next few releases.

**Reply by geiseri** [reply]
Sweet!  To be honest, what coder is now is much easier to deploy and manage for a small dev team than eclipse che.   If needing to build a container twice is the worst issue I have with coder you are doing awesome!

**Comment by matifali** [comment, marked as answer]
Please give https://github.com/coder/envbuilder a try. We also have sample templates to support devcontainer OOTB.

1. https://github.com/coder/coder/tree/main/examples/templates/devcontainer-docker
2. https://github.com/coder/coder/tree/main/examples/templates/devcontainer-kubernetes

**Reply by geiseri** [reply]
I gave it a whirl.  Exactly what I am looking for!  I did create a ticket for https://github.com/coder/envbuilder/issues/31 though but for other cases, it works exactly as advertised.
